File: benchmarks/oopsla25/subjects/971_getAddrSpec.c
#include <assert.h>
#include <fcntl.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <unistd.h>

char* f971_getAddrSpec(const char *email) {
    char *b1 = strchr(email, '<') + 1;
    char *b2 = strchr(b1, '>');
    size_t length = b2 - b1;
    char *addrSpec = malloc(length + 1);
    strncpy(addrSpec, b1, length);
    addrSpec[length] = '\0';
    return addrSpec;
}

int main(int argc, char *argv[]) {
  char my_string[10240] = {0};
  if (argc == 1) {
    read(fileno(stdin), my_string, 10240);
  } else {
    int fd = open(argv[1], O_RDONLY);
    read(fd, my_string, 10240);
    close(fd);
  }
  char *s = f971_getAddrSpec(my_string);
  printf("%s\n", s);
  return 0;
}

File: benchmarks/oopsla25/subjects/971_getAddrSpec.py
#!/usr/bin/env python3

import sys

def getAddrSpec(email):
  b1 = email.index('<',0)+1
  b2 = email.index('>',b1)
  return email[b1:b2]

if __name__ == '__main__':
  my_string = ""
  if len(sys.argv) == 1:
    my_string = sys.stdin.read()
  else:
    fd = open(sys.argv[1], 'r')
    my_string = fd.read()
    fd.close()
  
  print(getAddrSpec(my_string))
